app/controllers/pensions/unauthorisedPayments/SurchargeAmountController.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.pensions.unauthorisedPayments
+
+import config.{AppConfig, ErrorHandler}
+import controllers.pensions.routes.PensionsSummaryController
+import controllers.predicates.AuthorisedAction
+import controllers.predicates.TaxYearAction.taxYearAction
+import forms.{AmountForm, FormUtils}
+import models.mongo.PensionsCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.pensions.unauthorisedPayments.SurchargeAmountView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class SurchargeAmountController @Inject()(implicit val mcc: MessagesControllerComponents,
+                                          authAction: AuthorisedAction,
+                                          view: SurchargeAmountView,
+                                          appConfig: AppConfig,
+                                          pensionSessionService: PensionSessionService,
+                                          errorHandler: ErrorHandler,
+                                          clock: Clock) extends FrontendController(mcc) with I18nSupport with SessionHelper with FormUtils {
+
+
+  val amountForm: Form[BigDecimal] = AmountForm.amountForm(
+    emptyFieldKey = "unauthorisedPayments.surchargeAmount.error.noEntry",
+    wrongFormatKey = "unauthorisedPayments.surchargeAmount.error.incorrectFormat"
+  )
+
+
+  def show(taxYear: Int): Action[AnyContent] = (authAction andThen taxYearAction(taxYear)).async { implicit request =>
+    pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+      case Some(data) =>
+        if(data.pensions.unauthorisedPayments.surchargeQuestion.contains(true)){
```

Comment:
You could combine those lines to make a guide condition; for example:

```
 pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
      case Some(data) if data.pensions.unauthorisedPayments.surchargeQuestion.contains(true) =>
        data.pensions.unauthorisedPayments.surchargeAmount
          .map(value => Future.successful(Ok(view(amountForm.fill(value), taxYear))))
          .getOrElse(Future.successful(Ok(view(amountForm, taxYear))))
      case _ =>
        //TODO - redirect to CYA page once implemented
        Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
    }
```

Note that I used map instead of the Some / None match;  this might be more idiomatic or merely a personally preference. But merely a suggestion.

Reply:
Right - those two branches are redirecting to the same page now, but that will change. I missed that.

However, you would just need to add a second case, like this:

```
pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
      case Some(data) if data.pensions.unauthorisedPayments.surchargeQuestion.contains(true) =>
        data.pensions.unauthorisedPayments.surchargeAmount
          .map(value => Future.successful(Ok(view(amountForm.fill(value), taxYear))))
          .getOrElse(Future.successful(Ok(view(amountForm, taxYear))))
      case Some(_) =>
        //TODO - redirect to unauthorised payments question page once implemented
        Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
      case None =>
        //TODO - redirect to CYA page once implemented
        Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
    }
```

It compiles, with all tests green.